pyteal/ast/subroutine.py

```diff
@@ -299,15 +300,15 @@ class _OutputKwArgInfo:
 
     @staticmethod
     def from_dict(kwarg_info: dict[str, abi.TypeSpec]) -> Optional["_OutputKwArgInfo"]:
-        if kwarg_info is None or len(kwarg_info) == 0:
-            return None
-        elif len(kwarg_info) == 1:
-            key = list(kwarg_info.keys())[0]
-            return _OutputKwArgInfo(key, kwarg_info[key])
-        else:
-            raise TealInputError(
-                f"illegal conversion kwarg_info length {len(kwarg_info)}."
-            )
+        match list(kwarg_info.keys()):
```

Comment:
wow! match in action. so exciting!!!

Reply:
but isn't this going to blow up in the case that `kwarg_info is None` ?

Reply:
Hopefully the 1st `match` of many. 😅 

> isn't this going to blow up in the case that kwarg_info is None ?

I responded in https://github.com/algorand/pyteal/pull/308#discussion_r861360073 because the question arose in multiple places.